cocos/core/math/math-native-ext.ts

```diff
@@ -105,4 +106,12 @@ if (NATIVE) {
 
     Object.setPrototypeOf(jsb.Color.prototype, Color.prototype);
     Object.defineProperty(jsb.Color.prototype, 'type', { configurable: true, enumerable: true, writable: false, value: MathType.COLOR });
+
+    ccclass('cc.Vec4')(jsb.Vec4);
+    ccclass('cc.Vec3')(jsb.Vec3);
+    ccclass('cc.Vec2')(jsb.Vec2);
+    ccclass('cc.Mat4')(jsb.Mat4);
+    ccclass('cc.Mat3')(jsb.Mat3);
+    ccclass('cc.Color')(jsb.Color);
+    ccclass('cc.Quat')(jsb.Quat);
```

Comment:
Why need these code?

Reply:
Editor uses 'cc.Vec4' to identify the type,if these decorator is missing, Editor will fail to recognize the type, resulting in a serialization error

Reply:
The deserialization procedure needs to get the type information